test/e2e/tests/test_table.py

```diff
@@ -155,3 +267,391 @@ def test_enable_ttl(self, forum_table):
+
+        # Set server side encryption
+        cr["spec"] = {
+            "sseSpecification": {
+                "enabled": True,
+                "sseType": "KMS"
+            }
+        }
+
+        # Patch k8s resource
+        k8s.patch_custom_resource(ref, cr)
+
+        table.wait_until(
+            table_name,
+            table.sse_specification_matcher(True, "KMS"),
+            timeout_seconds=120,
+            interval_seconds=3,
+        )
+
+        time.sleep(60)
+
+        cr = k8s.wait_resource_consumed_by_controller(ref)
+        # Unset server side encryption
+        cr["spec"] = {
+            "sseSpecification": {
+                "enabled": False,
+            }
+        }
+
+        # Patch k8s resource
+        k8s.patch_custom_resource(ref, cr)
+
+        table.wait_until(
+            table_name,
+            table.sse_specification_matcher(False, ""),
+            timeout_seconds=120,
+            interval_seconds=3,
+        )
+
+    def test_update_class(self, forum_table):
+        (ref, res) = forum_table
+
+        table_name = res["spec"]["tableName"]
+
+        # Check DynamoDB Table exists
+        assert self.table_exists(table_name)
+
+        # Get CR latest revision
+        cr = k8s.wait_resource_consumed_by_controller(ref)
+
+        # Set table class
+        cr["spec"] = {
+            "tableClass": "STANDARD_INFREQUENT_ACCESS"
+        }
+
+        # Patch k8s resource
+        k8s.patch_custom_resource(ref, cr)
+
+        table.wait_until(
+            table_name,
+            table.class_matcher("STANDARD_INFREQUENT_ACCESS"),
+            timeout_seconds=180,
+            interval_seconds=3,
+        )
+
+        time.sleep(60)
+
+        cr = k8s.wait_resource_consumed_by_controller(ref)
+
+        # Set table class
+        cr["spec"] = {
+            "tableClass": "STANDARD"
+        }
+
+        # Patch k8s resource
+        k8s.patch_custom_resource(ref, cr)
+
+        table.wait_until(
+            table_name,
+            table.class_matcher("STANDARD"),
+            timeout_seconds=180,
+            interval_seconds=3,
+        )
+
+    def test_simple_create_gsi(self, office_table):
+        (ref, res) = office_table
+
+        table_name = res["spec"]["tableName"]
+
+        # Check DynamoDB Table exists
+        assert self.table_exists(table_name)
+
+        table.wait_until(
+            table_name,
+            table.gsi_matches([new_gsi_dict("office-per-city", "City", 5)]),
+            timeout_seconds=60*10,
+            interval_seconds=15,
+        )
+
+    def test_create_multi_gsi(self, office_table):
+        (ref, res) = office_table
+
+        table_name = res["spec"]["tableName"]
+
+        # Check DynamoDB Table exists
+        assert self.table_exists(table_name)
+
+        # Get CR latest revision
+        cr = k8s.wait_resource_consumed_by_controller(ref)
+
+        # Creating two more GSIs
+        cr["spec"] = {
```

Comment:
Do you mean to entirely overwrite `cr["spec"]` here?  why not just overwrite `cr["spec"]["attributeDefinitions"]` and `cr["spec"]["globalSecondaryIndexes"]`?

Reply:
This does a strategic merge, it doesn't remove the other fields :)
But i like your suggestion better, changing...